lib/puppet/pops/types/annotatable.rb

```diff
@@ -0,0 +1,30 @@
+module Puppet::Pops
+module Types
+
+KEY_ANNOTATIONS = 'annotations'.freeze
+
+module Annotatable
```

Comment:
# @api private
plus some comment 1-2 lines explaining what this is would be nice. "Behaviour common to all Pcore annotateable classes" or somesuch